pkg/daemons/agent/agent.go

```diff
@@ -200,11 +200,11 @@ func checkCgroups() (kubeletRoot, runtimeRoot string, hasCFS, hasPIDs bool) {
 		if err != nil {
 			return "", "", false, false
 		}
-		for _, c := range controllers {
-			switch c {
-			case "cpu":
+		for _, controller := range controllers {
+			switch {
```

Comment:
Can you add `controller` to the switch call? Then on the case lines, you can add the string you want matched and not need to do the eval in the case. Same for the switches below.

```go
switch controller {
case "cpu":
    // 
case "pids":
    ///
}
```